pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -426,3 +438,243 @@ func (vs *VSphere) Routes() (cloudprovider.Routes, bool) {
 func (vs *VSphere) ScrubDNS(nameservers, searches []string) (nsOut, srchOut []string) {
 	return nameservers, searches
 }
+
+// Attaches given virtual disk volume to the compute running kubelet.
+func (vs *VSphere) AttachDisk(vmDiskPath string, nodeName string) (diskID string, diskUUID string, err error) {
+	// Create context
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
the VM setup logic looks mostly identical in `AttachDisk`, `DetachDisk`, `CreateVolume` and `DeleteVolume`, can it be refactored?